### ISSUE TITLE
Steps that throw an error are not marked FAILED

### ISSUE THREAD

**Issue by gjw** [acuminous/yadda#223]
Weird and old, but relevant.

I'm trying to revive or fork marvin-js.  Marvin depends on yadda 0.16.0, everything is fine.  Update the dependencies, and updating yadda causes a major problem.  I bisected the failure point to between 0.17.10 and 0.18.0.  It seems that it's happening in lib/Macro.js, in Macro.interpret() (the situation remains the same from 18 -> current).

Somehow, wrapping the `result = fn...` in the try/catch is preventing actual errors from being listed as a failure.  Like,
```
  steps.when("I throw an error", function() {
    throw new Error('throwin\' it!');
  });
``` 
passes.

It seems likely to me that it's an interaction problem (else yadda has just been letting failing tests pass for several releases), but I don't have the faintest idea where to look next.  Ideas?

**Comment by cressie176** [comment]
Hi. Thanks for reporting. Looks like a bug in synchronous steps only.

**Comment by cressie176** [comment]
Just released 0.22.0. Please let me

**Comment by gjw** [comment]
Yup, that seems to have fixed it.

Thanks much!
